File: diario-classe-backend/src/main/java/com/projetointegrador/diarioclasse/entity/Observacao.java
package com.projetointegrador.diarioclasse.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tb_observacao")
public class Observacao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDate data;
    private String descricao;
    private String categoria;

//    @ManyToOne
//    @JoinColumn(name = "professor_id", nullable = false)
//    private Professor professor;

    @ManyToOne
    @JoinColumn(name = "aluno_id", nullable = false)
    private Aluno aluno;

    @ManyToOne
    @JoinColumn(name = "turma_id", nullable = false)
    private Turma turma;

    @ManyToOne
    @JoinColumn(name = "disciplina_id", nullable = false)
    private Disciplina disciplina;


    public Boolean isCategoriaComportamento() {
        return "comportamento".equalsIgnoreCase(this.categoria);
    }
}

File: diario-classe-frontend/src/models/Observacao.ts
import type {CategoriaObservacao} from "../enums/CategoriaObservacao.ts";

export interface Observacao {
  id: number;
  data: string;
  categoria: CategoriaObservacao;
  descricao: string;
  alunoId: number;
  alunoNome?: string;
  professorId: number;
  professorNome?: string;
  turmaId: number;
  turmaNome?: string;
  disciplinaId: number;
  disciplinaNome?: string;
}
